ctapipe/core/tool.py

```diff
@@ -170,23 +174,28 @@ def run(self, argv=None):
             Provenance().finish_activity(activity_name=self.name)
         except ToolConfigurationError as err:
             self.log.error(f'{err}.  Use --help for more info')
+            return 2  # wrong cmd line parameter
```

Comment:
will this prevent the `finally` block from executing?

Reply:
I think we should make sure it still runs, since that is where provenance is written (which can still be written if the tool failed, with the appropriate status recorded). That is perhaps useful if we use the provenance info to track if jobs complete correctly or not

Reply:
I think having a single "return" point is better form anyhow (minimize complex branching), so just refactoring this to set a `return_value` and returning it at the end should suffice.

Reply:
No, finally will still execute. See https://docs.python.org/3.7/tutorial/errors.html#defining-clean-up-actions

> A finally clause is always executed before leaving the try statement, whether an exception has occurred or not. When an exception has occurred in the try clause and has not been handled by an except clause (or it has occurred in an except or else clause), it is re-raised after the finally clause has been executed. The finally clause is also executed “on the way out” when any other clause of the try statement is left via a break, continue or return statement.

Reply:
Ah, that's not at all what I expected! Ok, it's fine then! I had to actually try it myself.